File: Leetcode/cpp/House Robber.cc
class Solution {
public:
    int rob(vector<int>& nums) {
        int A = 0; // (x - 1)th house is robbed
        int B = 0; // (x - 1)th house is not robbed
        for (auto i: nums) {
            int tmp = B;
            B = max(A, B);
            A = tmp + i;
        }
        return max(A, B);
    }
};

File: Leetcode/Algorithm/java/House Robber.java
public class Solution {
    public int rob(int[] nums) {
        int A = 0;
        int B = 0;
        for (int num: nums) {
            int _A = num + B;
            int _B = Math.max(A, B);
            
            A = _A;
            B = _B;
        }
        return Math.max(A, B);
    }
}
